src/main/java/org/spongepowered/api/event/entity/living/player/PlayerInteractEvent.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.spongepowered.api.event.entity.living.player;
+
+import com.flowpowered.math.vector.Vector3f;
+import com.google.common.base.Optional;
+import org.spongepowered.api.entity.EntityInteractionType;
+import org.spongepowered.api.entity.player.Player;
+import org.spongepowered.api.event.entity.living.human.HumanInteractEvent;
+
+/**
+ * Called when a {@link Player} interacts with the world.
+ */
+public interface PlayerInteractEvent extends PlayerEvent, HumanInteractEvent {
+
+    /**
+     * Gets the {@link EntityInteractionType} that the player used.
+     *
+     * @return The type of click
+     */
+    EntityInteractionType getInteractionType();
+
+    /**
+     * Gets the clicked position of this interact event.
+     *
+     * <p>This may not always be available, in which case
+     * {@link Optional#absent()} is returned. Specifically, when a player
```

Comment:
Currently, the location is sent for all entities (although it's only used for `AmorStand`s)